Visual Studio compilation documentation

NlightNFotis opened the pull request:
This PR serves two purposes:

1. Add documentation on building with Visual Studio on Windows to the `COMPILING.md` file.
2. Do some further minor cleanup on top of #6109 and implement some suggestions from the
    review comments there.


<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.

If your PR fixes a bug, include the regression test(s) in the same commit as the bug fix. Else, keep commits small and orthogonal, possibly placing tests in commits of their own.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

NlightNFotis committed:
Add instructions on building on Windows with Visual Studio.

TGWDB commented on COMPILING.md:
```diff
@@ -114,6 +114,25 @@ files.
+   to something like `x64-Debug (Default)` or whatever the equivalent is
+   for your system.
+
+The above instructions hae been tested against Visual Studio 2019.
```
Above says VS 2017, which version is it?

NlightNFotis replied:
I have tested it with Visual Studio 2019, but 2017 is the first version that added support for this.

Do I convey this accurately through the text?

thomasspriggs replied:
I am OK with the two different version numbers above, as the versions match the facts. Maybe add 2019 to the section title though.

tautschnig commented on COMPILING.md:
```diff
@@ -114,6 +114,25 @@ files.
 
+## Compilation on Windows using Visual Studio
+
+Follow these instructions to work on Windows with Visual Studio:
```
It would be useful to make clear whether that's on top of the above instructions, or a replacement of (some of) the steps.

tautschnig commented on COMPILING.md:
```diff
@@ -114,6 +114,25 @@ files.
+
+Follow these instructions to work on Windows with Visual Studio:
+
+1. First, clone the `cbmc` repository locally.
```
Fair call, but other build instructions don't suggest that. We should be consistent about that.

thomasspriggs replied:
:pick: This isn't a Visual Studio specific instruction. Visual Studio 2019 features git integration. So it would be great to include instructions on cloning CBMC using that integration - [Visual Studio git documentation](https://docs.microsoft.com/en-us/visualstudio/version-control/git-with-visual-studio?view=vs-2019#clone-an-existing-git-repository)

tautschnig commented on COMPILING.md:
```diff
@@ -114,6 +114,25 @@ files.
+Follow these instructions to work on Windows with Visual Studio:
+
+1. First, clone the `cbmc` repository locally.
+2. Open the `cbmc` folder into Visual Studio. Visual Studio 2017 and
```
English isn't my first language, so take this with a grain of salt: I think it should be "in," not "into." Perhaps someone whose first language _is_ English can do a proof read?

TGWDB replied:
Yes, "into" -> "in"

tautschnig commented on COMPILING.md:
```diff
@@ -114,6 +114,25 @@ files.
+   to something like `x64-Debug (Default)` or whatever the equivalent is
+   for your system.
+
+The above instructions hae been tested against Visual Studio 2019.
```
Nit pick: s/hae/have/

tautschnig commented on src/goto-checker/symex_coverage.cpp:
```diff
@@ -14,6 +14,7 @@ Date: March 2016
 #include "symex_coverage.h"
 
 #include <chrono>
+#include <ctime>
```
Thank you for fixing this! Would you mind moving it to a PR of its own, I'll happily add my approval as soon as I get to see the PR. And: my apologies for messing this up.

NlightNFotis replied:
Yeah, no problem - I will spin this out to its own PR now.

Also, yeah it's hard to notice subtle breakages like this - I only discovered it yesterday when I played around with VS on windows to write these docs.

thomasspriggs commented on COMPILING.md:
```diff
@@ -114,6 +114,25 @@ files.
    complete set of built binaries can be found in `./build/bin` once the build
    is complete.
 
+## Compilation on Windows using Visual Studio
```
:pick: I don't think this section should really be just about *compilation* with Visual Studio as much as its use as an IDE. As such I suggest that this new section be moved be moved just above the existing section on "WORKING WITH ECLIPSE" and that both sections be made subsections of a new "Working on CBMC using IDEs" section.

:bulb: We could also add a new "Working with CLion" section as a later enhancement (as a separate piece of work in a potential follow-up PR).

thomasspriggs reviewed (commented):
Needs a rebase to get rid of the commits which have already been merged as separate PRs.

NlightNFotis committed:
Implement suggestions from #6109 and some further cleanup.

Implement some suggestions presented as comments in #6109 (we
had to merge the PR before we could get a chance to incorporate
them, so we are compensating now) and do some other minor
cleanup of the COMPILING.md file.

NlightNFotis commented:
Hi @tautschnig, can you please re-review the second commit in this PR?

We had to merge #6109 before we could incorporate your suggested fixes, so this is now including them.
